mgmt/src/processor/confbuild.rs

```diff
@@ -204,55 +204,56 @@ fn build_vpc_internal_config(
     Ok(())
 }
 
-/// Get the config for the (a) loopback interface
-fn get_loopback_interface_config(
-    external: &ExternalConfig,
-) -> Result<&InterfaceConfig, ConfigError> {
-    external
+struct VtepInfo {
+    ip: UnicastIpv4Addr,
+    mac: SourceMac,
+}
+
+fn get_vtep_info(external: &ExternalConfig) -> Result<VtepInfo, ConfigError> {
```

Comment:
Wouldn't it be easier to simply validate the vtep interface in https://github.com/githedgehog/dataplane/blob/main/mgmt/src/models/internal/interfaces/interface.rs#L167

... to avoid needing to iterate over all of the interfaces to do that validation?
The `InterfaceConfig::validate()` is executed once per interface from https://github.com/githedgehog/dataplane/blob/main/mgmt/src/models/external/gwconfig/mod.rs#L34

That is the place where I would add the restriction for Ipv4 only (okay and unicast) so that both the gRPC conversion and the rest of code can be dual version.

Reply:
Also, the building of the configuration is fallible. However, it logically occurs after having validated it. So, if we want to retain that structure, I'd move the validation there.

Reply:
Because the interface set is still a list here, I would still have to do the check dynamically since we do not want to count on validate doing the right thing as the code changes.  We either encode the checks we did in the type system or we check dynamically where there is a possible error.  

I am glad to add a check to the validator as well though so that we fail earlier and move these to internal errors.
I will make that fix.

Reply:
ok!

Reply:
Moved the real checking code into the validator.  Any checks needed for type conversion/extraction in confbuild now return `InternalFailure`